clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2119,17 +2085,16 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
     return true;
   }
 
-  // FIXME: Refactor accessor handling. There is a discrepancy in how
-  // inherited accessors are handled.
-  bool handleSyclAccessorType(const CXXRecordDecl *, const CXXBaseSpecifier &BS,
-                              QualType FieldTy) final {
+  bool handleSyclSpecialType(const CXXRecordDecl *, const CXXBaseSpecifier &BS,
```

Comment:
I think you will need to make the same modifications you made in above function in this function as well, in order to differentiate required additional accessor handling from streams/samplers